pkg/controller/datavolume/controller-base.go

```diff
@@ -323,18 +346,123 @@ func (r ReconcilerBase) syncUpdate(log logr.Logger, syncRes *dataVolumeSyncResul
 	return nil
 }
 
-func (r ReconcilerBase) handlePrePopulation(dv *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim) {
+func (r *ReconcilerBase) handleStaticVolume(syncRes *dataVolumeSyncResult, log logr.Logger) error {
+	if _, ok := syncRes.dvMutated.Annotations[cc.AnnCheckStaticVolume]; !ok {
+		return nil
+	}
+
+	if syncRes.pvc == nil {
+		volumes, err := r.getAvailableVolumesForDV(syncRes, log)
+		if err != nil {
+			return err
+		}
+
+		if len(volumes) == 0 {
+			log.Info("No PVs for DV")
+			return nil
+		}
+
+		if err := r.handlePvcCreation(log, syncRes, func(_ *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim) error {
+			bs, err := json.Marshal(volumes)
+			if err != nil {
+				return err
+			}
+			cc.AddAnnotation(pvc, cc.AnnPersistentVolumeList, string(bs))
+			return nil
+		}); err != nil {
+			return err
+		}
+
+		// set result to make sure callers don't do anything else in sync
+		syncRes.result = &reconcile.Result{}
+		return nil
+	}
+
+	volumeAnno, ok := syncRes.pvc.Annotations[cc.AnnPersistentVolumeList]
+	if !ok {
+		// etiher did not create the PVC here OR bind to expected PV succeeded
+		return nil
+	}
+
+	if syncRes.pvc.Spec.VolumeName == "" {
+		// set result to make sure callers don't do anything else in sync
+		syncRes.result = &reconcile.Result{}
+		return nil
+	}
+
+	var volumes []string
+	if err := json.Unmarshal([]byte(volumeAnno), &volumes); err != nil {
+		return err
+	}
+
+	for _, v := range volumes {
+		if v == syncRes.pvc.Spec.VolumeName {
+			pvcCpy := syncRes.pvc.DeepCopy()
+			// handle as "populatedFor" going foreward
+			cc.AddAnnotation(pvcCpy, cc.AnnPopulatedFor, syncRes.dvMutated.Name)
+			delete(pvcCpy.Annotations, cc.AnnPersistentVolumeList)
+			if err := r.updatePVC(pvcCpy); err != nil {
+				return err
+			}
+			syncRes.pvc = pvcCpy
+			return nil
+		}
+	}
+
+	// delete the pvc and hope for better luck...
+	pvcCpy := syncRes.pvc.DeepCopy()
```

Comment:
why copy is needed here? and why set `syncRes.pvc = pvcCpy` after the deletion instead of setting it to `nil`?

Reply:
Same reasoning as above, what if the delete call mutates the variable in some way and fails?  I don't think it does but I think you get my drift